custody-fireblocks/src/main/kotlin/com/stablecoin/custody/fireblocks/infrastructure/security/FireblocksWebhookAuthenticationFilter.kt

```
@@ -0,0 +1,129 @@
+            log.warn("Signature verification failed: {}", e.message)
+            false
+        }
+
+    private fun isWithinReplayWindow(bodyBytes: ByteArray): Boolean =
+        try {
+            val tree = objectMapper.readTree(bodyBytes)
+            val createdAt = tree.get("createdAt")?.asLong() ?: return false
+            val eventTime = Instant.ofEpochMilli(createdAt)
+            val age = Duration.between(eventTime, Instant.now())
+            age <= REPLAY_WINDOW
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)